aspnet/mvc/models/formatting.rst

```diff
@@ -1,8 +1,223 @@
+
+Special Case Formatters
+^^^^^^^^^^^^^^^^^^^^^^^
+
+Some special cases are implemented using built-in formatters. By default, ``string`` return types will be formatted as `text/plain` (`text/html` if requested via ``Accept`` header). This behavior can be removed by removing the ``TextOutputFormatter``. You remove formatters in the ``Configure`` method in *Startup.cs* (shown below). Actions that have a model object return type will return a 204 No Content response when returning ``null``. This behavior can be removed by removing the ``HttpNoContentOutputFormatter``. The following code removes the ``TextOutputFormatter`` and `HttpNoContentOutputFormatter``.
+
+.. code-block:: c#
+
+  services.AddMvc(options =>
+  {
+    options.OutputFormatters.RemoveType<TextOutputFormatter>();
+    options.OutputFormatters.RemoveType<HttpNoContentOutputFormatter>();
+  });
+  
+Without the ``TextOutputFormatter``, ``string`` return types return 406 Not Acceptable.
+
+Without the ``HttpNoContentOutputFormatter``, null objects are formatted using the configured formatter. For example, the JSON formatter will simply return a response with a body of ``null``, while the XML formatter will return an empty XML element with the attribute ``xsi:nil="true"`` set.
+
+Response Format URL Mappings
+----------------------------
+
+Clients can request a particular format as part of the URL, such as in the query string or part of the path, or by using a format-specific file extension such as .xml or .json. The mapping from request path should be specified in the route the API is using. For example:
+
+.. code-block:: c#
+
+  [FormatFilter]
+  public class ProductsController
+  {
+    [Route("[controller]/[action]/[id].{format?}")]
+    public Product GetById(int id)
+  
+This route would allow the requested format to be specified as an optional file extension. The ``[FormatFilter]`` attribute checks for the existence of the format value in the ``RouteData`` and will map the response format to the appropriate formatter when the response is created.
+
+.. list-table:: Examples
+  :header-rows: 1
+  
+  * - Route
+    - Formatter
+  * - ``/products/GetById/5`` 
+    - The default output formatter
+  * - ``/products/GetById/5.json``
+    - The JSON formatter (if configured)
+  * - ``/products/GetById/5.xml``
+    - The XML formatter (if configured)
+
+Recommendations
+---------------
+
+There are many options when it comes to returning data from API actions. Simple return types may be appropriate for prototype code. We recommend APIs that will be used in production use ``IActionResult`` as your return type for data. Returning ``IActionResult`` allows you to handle error states and different HTTP status codes explicitly.
```

Comment:
Isn't using IActionResult already discussed and recommended earlier? I think this entire section can be removed.